### PR TITLE
Support type cast in group by

### PR DESCRIPTION
**Pull request by clemvnt**
Hello,

This is the initial step towards supporting type cast in group by.

Related to https://github.com/OData/AspNetCoreOData/pull/1182.

### Issues

https://github.com/OData/AspNetCoreOData/issues/1180

### Description

For the following group by : `groupby((MyAddress/Fully.Qualified.Namespace.HomeAddress/HomeNO))`

The response was :

```json
[
  {
    "homeNO": "Value1"
  },
  {
    "homeNO": "Value2"
  }
]
```

Now, the response is :

```json
[
  {
    "myAddress": {
        "homeNO": "Value1"
    }
  },
  {
    "myAddress": {
        "homeNO": "Value2"
    }
  }
]
```

### Checklist

- [X] *Test cases added*
- [X] *Build and test with one-click build and test script passed*

### CODE REVIEW

**Comment by clemvnt** [comment]
@microsoft-github-policy-service agree

**Review by xuzhg — APPROVED** [review]

File: test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Extensions/Binders/ApplyBinderTests.cs [hunk 1]
```diff
@@ -757,6 +757,58 @@ public void BindVirtualPropertiesAfterCollapseReturnsApplyClause()
             Assert.Equal(AggregationMethod.VirtualPropertyCount, aggExp.Method);
         }
 
+        [Fact]
+        public void BindApplyWithGroupByAndTypeCastShouldReturnApplyClause()
```
Comment (wachugamaina) [inline test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Extensions/Binders/ApplyBinderTests.cs:761]: These two tests seem very similar 
Could they not be made one parameterized test ?

Comment (clemvnt) [reply test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Extensions/Binders/ApplyBinderTests.cs:761]: The assertions are not the same. I don't really see how to merge these tests into a parameterizable test without having to use conditions (which makes the code less understandable).

**Comment by clemvnt** [comment]
Is there something blocking this PR?

**Review by habbes — APPROVED** [review]

**Review by ElizabethOkerio — APPROVED** [review]

**Pull request merged**